torchdynamo/variables/tensor.py

```diff
@@ -269,6 +277,15 @@ def var_getattr(self, tx, name):
         if name == "__class__":
             return TorchVariable(self.python_type(), **options)
 
+        # Add a guard for type matching, these guards are checked before tensor guards
+        # In some cases, a <tensor>.<attr> guard can be evaluated first, and break if
+        # <tensor> is later changed another type
+        try:
+            if result is not None:
```

Comment:
perhaps we need `and self.source is not None`

Reply:
but isn't control flow with exceptions "pythonic"? jk done.